packages/preview-middleware-client/src/cpe/changes/service.ts

```diff
@@ -353,12 +438,54 @@ export class ChangeService {
             this.eventStack.splice(eventIndex, 1);
             if (Array.isArray(allCommands) && allCommands.length === 0) {
                 this.pendingChanges = [];
+                this.pendingConfigChangeMap = new Map();
                 await this.fetchSavedChanges();
             }
+
+            // Notify to update the ui for configuration changes.
+            const configurationChanges = this.pendingChanges?.filter((item) => item.kind === 'configuration');
+            if (configurationChanges.length) {
+                const event = new CustomEvent(STACK_CHANGE_EVENT, {
+                    detail: {
+                        controls: (configurationChanges as PendingConfigurationChange[]).reduce((acc, item) => {
+                            const controls = (item.controlIds || [])
+                                .map((id: string) => {
+                                    return getControlById(id);
+                                })
+                                .filter((item) => item?.getId()) as UI5Element[];
```

Comment:
Is this to handle the cases where the control is not found?

Reply:
as discussed switched to isA